src/js/utils/date-utils.js

```diff
@@ -0,0 +1,289 @@
+             *
+             */
+            if (typeof format !== undefined) {
+                displayformat = DATE_FORMATS[format];
+            }
+            if (typeof format === undefined || format === undefined || format === 'None') {
+                displayformat = DATE_FORMATS.DEFAULT_DATE_TIME_FORMAT;
+            }
+            displaytime = localizedtime.locale(displaylanguage).format(displayformat);
+
+            /**
+             * This is a final error bailout -
+             * will default to UTC/en-US display in default format in the case of
+             * an unresolvable language or timezone
+             *
+             */
+            if (displaytime.length > 0 && displaytime !== 'Invalid date') {
+                return displaytime;
+            }
+            displaytime = localizedtime.tz('UTC').locale('en').format(displayformat);
+            // console.log('DateUtil Localization Error');
+            return displaytime;
+        };
+
+        /**
+         *
+         * This should take a context dict, with an additional (and optional)
+         * modifier_string (e.g. 'due', 'deadline:', etc.)
+         *
+         * This will take all the contexts and render a localized date/time in the
+         * standard format of the developer's choice.
+         *
+         * @param {Object.<string, string>} context
+         *
+         *      datetime: <string> The UTC formatted datetime string from the server,
+         *      modifier: <string> (optional) 'due', 'deadline:', etc.,
+         *      timezone: <string> (optional) the user set time zone preference, or 'None' if unset
+         *          This should be in a format like 'Region/Locale', e.g. 'Atlantic/Azores',
+         *      language: <string> (optional), the user set language, an ISO 639-1 language string,
+         *      format: <string> (optional), a format constant
+         *      }
+         *
+         * @returns {string}
+         *
+         * Note, for clarity I broke this up into several functions, but this could be a one-liner:
+         * display_time.moment(raw_datetime).tz('UTC').tz(display_zone).locale(display_locale).format('ll HH[:]mm z');
+         */
+        localize = function(context) {
+            /**
+             * made a little longer than necessary for clarity
+             *
+             */
+            var localizedtime;
+            var displaytime;
+            var displaystring;
+
+            if (context.datetime === undefined || context.datetime === '') {
+                return '';
+            }
+            localizedtime = LocalizedTime(
+                DateUtils.DateObject(context.datetime),
+                UserTimeZone(context.timezone)
+                );
+            displaytime = TimeDisplay(localizedtime, context.language, context.format);
+
+            if (context.modifier === undefined || context.modifier.length === 0) {
+                return displaytime;
+            }
+            /**
```

Comment:
A normal block comment (just `/*` `*/`) should be fine here, we only use the JSDoc style for function definitions.